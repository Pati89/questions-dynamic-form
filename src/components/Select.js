import PropTypes from "prop-types";
import styles from "../styles/components/Select.module.css";

const Select = (props) => {
  const { field, onChange, value } = props;

  const { name, label, options } = field;

  return (
    <div className={styles.selectField} data-testid="select">
      <label htmlFor={name}>{label}</label>
      <select
        {...props}
        type="text"
        id={`${name}-select`}
        name={name}
        onChange={(event) => onChange(event, name)}
        value={value || ""}
        data-testid="select-field"
      >
        <option value="" data-testid="select-option">
          --Please choose an option--
        </option>
        {options?.map((item, index) => (
          <option key={index} value={item} data-testid="select-option">
            {item}
          </option>
        ))}
      </select>
    </div>
  );
};

Select.propTypes = {
  field: PropTypes.object,
};

export default Select;
