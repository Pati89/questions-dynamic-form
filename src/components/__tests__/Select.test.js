import Select from "../Select";
import { render, screen } from "@testing-library/react";

const props = {
  field: {
    name: "test_field",
    label: "Test Select Field",
    type: "dropdown",
    options: ["Option1", "Option2", "Option3"],
  },
};

describe("Select", () => {
  test("should render a Select", () => {
    render(<Select {...props} />);
    const select = screen.getByTestId("select");
    expect(select).toBeTruthy();
    expect(select).toHaveTextContent("Test Select Field");
  });

  test("should pass name prop", () => {
    render(<Select {...props} />);
    const selectField = screen.getByTestId("select-field");
    expect(selectField).toBeTruthy();
    expect(selectField).toHaveProperty("name", "test_field");
  });

  test("should render options", () => {
    render(<Select {...props} />);
    const selectOption = screen.getAllByTestId("select-option");
    expect(selectOption).toHaveLength(4);
  });

  test("should render a Select match snapshot", () => {
    render(<Select {...props} />);
    const select = screen.getByTestId("select");
    expect(select).toMatchSnapshot();
  });
});
