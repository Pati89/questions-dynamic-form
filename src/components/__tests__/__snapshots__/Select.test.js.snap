// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select should render a Select match snapshot 1`] = `
<div
  class="selectField"
  data-testid="select"
>
  <label
    for="test_field"
  >
    Test Select Field
  </label>
  <select
    data-testid="select-field"
    field="[object Object]"
    id="test_field-select"
    name="test_field"
    type="text"
  >
    <option
      data-testid="select-option"
      value=""
    >
      --Please choose an option--
    </option>
    <option
      data-testid="select-option"
      value="Option1"
    >
      Option1
    </option>
    <option
      data-testid="select-option"
      value="Option2"
    >
      Option2
    </option>
    <option
      data-testid="select-option"
      value="Option3"
    >
      Option3
    </option>
  </select>
</div>
`;
